{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from transformers import BertForMaskedLM, BertTokenizerFast\n",
    "import os\n",
    "\n",
    "model_path = r\"C:\\Users\\qja19\\Desktop\\issue_bert\\10000i_5000s_large_40ep\"\n",
    "tokenizer = BertTokenizerFast.from_pretrained(model_path)\n",
    "model = BertForMaskedLM.from_pretrained(os.path.join(model_path, \"checkpoint-last\"))\n",
    "\n",
    "REPO_NAME = 'issueBERT-large' # ex) 'my-bert-fine-tuned'\n",
    "AUTH_TOKEN = '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' # <https://huggingface.co/settings/token>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\site-packages\\huggingface_hub\\hf_api.py:1001: FutureWarning: `create_repo` now takes `token` as an optional positional argument. Be sure to adapt your code!\n",
      "  warnings.warn(\n",
      "Cloning https://huggingface.co/gbkwon/issueBERT-large into local empty directory.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0b1f142aba7c4bb396a97168530ecb46",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Download file pytorch_model.bin:   0%|          | 8.00k/1.25G [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3f6c3a0379e94f1fa31b2c56763cb7d0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Clean file pytorch_model.bin:   0%|          | 1.00k/1.25G [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "PermissionError",
     "evalue": "[WinError 5] 액세스가 거부되었습니다: 'C:\\\\Users\\\\qja19\\\\AppData\\\\Local\\\\Temp\\\\tmp8d77xuem\\\\.git\\\\objects\\\\42\\\\d26a8e0b12b1731dc84099926af71466f6882b'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mPermissionError\u001b[0m                           Traceback (most recent call last)",
      "\u001b[1;32mc:\\Users\\qja19\\Desktop\\issue_bert\\upload.ipynb Cell 2\u001b[0m line \u001b[0;36m2\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m \u001b[39m## Upload to Huggingface Hub\u001b[39;00m\n\u001b[1;32m----> <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m model\u001b[39m.\u001b[39;49mpush_to_hub(\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=2'>3</a>\u001b[0m     REPO_NAME, \n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=3'>4</a>\u001b[0m     use_temp_dir\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m, \n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=4'>5</a>\u001b[0m     use_auth_token\u001b[39m=\u001b[39;49mAUTH_TOKEN\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/qja19/Desktop/issue_bert/upload.ipynb#W1sZmlsZQ%3D%3D?line=5'>6</a>\u001b[0m )\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\site-packages\\transformers\\modeling_utils.py:2301\u001b[0m, in \u001b[0;36mPreTrainedModel.push_to_hub\u001b[1;34m(self, repo_path_or_name, repo_url, use_temp_dir, commit_message, organization, private, use_auth_token, max_shard_size, **model_card_kwargs)\u001b[0m\n\u001b[0;32m   2299\u001b[0m \u001b[39m# Clean up! Clean up! Everybody everywhere!\u001b[39;00m\n\u001b[0;32m   2300\u001b[0m \u001b[39mif\u001b[39;00m use_temp_dir:\n\u001b[1;32m-> 2301\u001b[0m     shutil\u001b[39m.\u001b[39;49mrmtree(repo_path_or_name)\n\u001b[0;32m   2303\u001b[0m \u001b[39mreturn\u001b[39;00m url\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:750\u001b[0m, in \u001b[0;36mrmtree\u001b[1;34m(path, ignore_errors, onerror)\u001b[0m\n\u001b[0;32m    748\u001b[0m     \u001b[39m# can't continue even if onerror hook returns\u001b[39;00m\n\u001b[0;32m    749\u001b[0m     \u001b[39mreturn\u001b[39;00m\n\u001b[1;32m--> 750\u001b[0m \u001b[39mreturn\u001b[39;00m _rmtree_unsafe(path, onerror)\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:615\u001b[0m, in \u001b[0;36m_rmtree_unsafe\u001b[1;34m(path, onerror)\u001b[0m\n\u001b[0;32m    613\u001b[0m         onerror(os\u001b[39m.\u001b[39mpath\u001b[39m.\u001b[39mislink, fullname, sys\u001b[39m.\u001b[39mexc_info())\n\u001b[0;32m    614\u001b[0m         \u001b[39mcontinue\u001b[39;00m\n\u001b[1;32m--> 615\u001b[0m     _rmtree_unsafe(fullname, onerror)\n\u001b[0;32m    616\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m    617\u001b[0m     \u001b[39mtry\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:615\u001b[0m, in \u001b[0;36m_rmtree_unsafe\u001b[1;34m(path, onerror)\u001b[0m\n\u001b[0;32m    613\u001b[0m         onerror(os\u001b[39m.\u001b[39mpath\u001b[39m.\u001b[39mislink, fullname, sys\u001b[39m.\u001b[39mexc_info())\n\u001b[0;32m    614\u001b[0m         \u001b[39mcontinue\u001b[39;00m\n\u001b[1;32m--> 615\u001b[0m     _rmtree_unsafe(fullname, onerror)\n\u001b[0;32m    616\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m    617\u001b[0m     \u001b[39mtry\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:615\u001b[0m, in \u001b[0;36m_rmtree_unsafe\u001b[1;34m(path, onerror)\u001b[0m\n\u001b[0;32m    613\u001b[0m         onerror(os\u001b[39m.\u001b[39mpath\u001b[39m.\u001b[39mislink, fullname, sys\u001b[39m.\u001b[39mexc_info())\n\u001b[0;32m    614\u001b[0m         \u001b[39mcontinue\u001b[39;00m\n\u001b[1;32m--> 615\u001b[0m     _rmtree_unsafe(fullname, onerror)\n\u001b[0;32m    616\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m    617\u001b[0m     \u001b[39mtry\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:620\u001b[0m, in \u001b[0;36m_rmtree_unsafe\u001b[1;34m(path, onerror)\u001b[0m\n\u001b[0;32m    618\u001b[0m             os\u001b[39m.\u001b[39munlink(fullname)\n\u001b[0;32m    619\u001b[0m         \u001b[39mexcept\u001b[39;00m \u001b[39mOSError\u001b[39;00m:\n\u001b[1;32m--> 620\u001b[0m             onerror(os\u001b[39m.\u001b[39;49munlink, fullname, sys\u001b[39m.\u001b[39;49mexc_info())\n\u001b[0;32m    621\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m    622\u001b[0m     os\u001b[39m.\u001b[39mrmdir(path)\n",
      "File \u001b[1;32mc:\\Users\\qja19\\Anaconda3\\envs\\bert\\lib\\shutil.py:618\u001b[0m, in \u001b[0;36m_rmtree_unsafe\u001b[1;34m(path, onerror)\u001b[0m\n\u001b[0;32m    616\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m    617\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m--> 618\u001b[0m         os\u001b[39m.\u001b[39;49munlink(fullname)\n\u001b[0;32m    619\u001b[0m     \u001b[39mexcept\u001b[39;00m \u001b[39mOSError\u001b[39;00m:\n\u001b[0;32m    620\u001b[0m         onerror(os\u001b[39m.\u001b[39munlink, fullname, sys\u001b[39m.\u001b[39mexc_info())\n",
      "\u001b[1;31mPermissionError\u001b[0m: [WinError 5] 액세스가 거부되었습니다: 'C:\\\\Users\\\\qja19\\\\AppData\\\\Local\\\\Temp\\\\tmp8d77xuem\\\\.git\\\\objects\\\\42\\\\d26a8e0b12b1731dc84099926af71466f6882b'"
     ]
    }
   ],
   "source": [
    "## Upload to Huggingface Hub\n",
    "model.push_to_hub(\n",
    "    REPO_NAME, \n",
    "    use_temp_dir=True, \n",
    "    use_auth_token=AUTH_TOKEN\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tokenizer.push_to_hub(\n",
    "    REPO_NAME, \n",
    "    use_temp_dir=True, \n",
    "    use_auth_token=AUTH_TOKEN\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bert",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
